src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala

```diff
@@ -0,0 +1,260 @@
+package org.clulab.habitus.utils
+
+import org.clulab.habitus.variables.EntityDistFreq
+
+import org.clulab.odin.{Attachment, Mention}
+import org.clulab.processors.Document
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+trait Context extends Attachment {
+  def getArgValuePairs() = this.getClass.getDeclaredFields.toList
+    .map(arg => {
+      arg.setAccessible(true)
+      (arg.getName, arg.get(this))
+    })
+
+  def getTSVContextHeader() = {
+    getArgValuePairs().map(_._1).mkString("\t")
+  }
+
+  def getTSVContextString() = {
+    getArgValuePairs().map(_._2).mkString("\t")
+  }
```

Comment:
It *seems* like the argValuePairs are generic enough.  Someone else will want CSV and someone else will somehow slip tabs or commas into the data so that the output is messed up.  It could be better to handle that part in the specific printers.